app-modules/alert/graphql/alert.graphql

```diff
@@ -27,22 +27,56 @@ type Alert @model(class: "AdvisingApp\\Alert\\Models\\Alert") {
     deleted_at: DateTime
 }
 
-extend type Query {
-    "Get an alert by its primary key."
-    alert("Search by primary key." id: UUID! @whereKey): Alert
-        @find
-        @canResolved(ability: "view")
+input AlertConcernsQuery {
+    student: StudentQuery
+    prospect: ProspectQuery
+}
 
-    "List multiple alerts." #TODO: filtering
-    alerts: [Alert!]! @paginate @softDeletes @canModel(ability: "viewAny")
+input AlertsQuery {
+    id: UUID
+    concern: AlertConcernsQuery @morphToRelation
+    concern_id: EducatableId
+    concern_type: EducatableType
+    description: String
+    severity: AlertSeverity
+    status: AlertStatus
+    suggested_intervention: String
+    created_at: DateTime
+    updated_at: DateTime
+    deleted_at: DateTime
 }
 
-input CreateAlertInput {
-    "The id of the concern of the alert."
-    concern_id: UUID! @rules(apply: ["required"])
+type AlertQueries {
+    "Find a single alert by an identifying attribute."
+    find(
+        "The value of the attribute to match."
+        id: UUID! @whereKey @rules(apply: ["required", "uuid", "exists:alerts"])
+    ): Alert @find @softDeletes @canResolved(ability: "view")
+
+    "List multiple alerts."
+    list(where: AlertsQuery @searchBy): [Alert!]!
+        @paginate
+        @softDeletes
+        @canModel(ability: "viewAny")
+}
 
-    "The type of concern of the alert."
-    concern_type: String! @rules(apply: ["required", "in:student,prospect"])
+extend type Query {
+    alert: AlertQueries! @namespaced
+}
+
+input CreateAlertInput {
+    "The Concern related to the alert."
+    concern_id: EducatableId!
+        @rules(
+            apply: [
+                "required"
+                "AdvisingApp\\Alert\\Rules\\ConcernIdExistsRules"
```

Comment:
This must have been a typo when I created the Interaction one and you copied mine. But I think we should have the class name be singular. `ConcernIdExistsRule`